addons/uom/models/uom_uom.py

```diff
@@ -107,7 +100,7 @@ def write(self, values):
         return super(UoM, self).write(values)
 
     def unlink(self):
-        if self.filtered(lambda uom: uom.measure_type == 'working_time'):
+        if self.filtered(lambda uom: uom.category_id in (self.env.ref('uom.product_uom_categ_unit'), self.env.ref('uom.uom_categ_wtime'))):
```

Comment:
I don't get this logic.  (it was wrong before, I guess)  I can delete a metric ton, but I can not delete a year or a box of 25 pieces?

Reply:
I didn't adapt the code correctly. Thanks for the catch. We can't delete an uom with the category unit or time if it's the reference for that category. I changed it now.